sklearn/decomposition/pca.py

```diff
@@ -183,9 +183,13 @@ class PCA(_BasePCA):
 
         .. versionadded:: 0.18.0
 
-    random_state : int or RandomState instance or None (default None)
+    random_state : int, RandomState instance or None, optional (default None)
         Pseudo Random Number generator seed control. If None, use the
         numpy.random singleton. Used by svd_solver == 'arpack' or 'randomized'.
```

Comment:
Seems like there's redundant documentation in some of these strings. Would it be possible for you to go through these cases (typically where it includes "Used by x == y") and remove the excess documentation and move the "Used by" line to the end? Like here it should be:

```
If int, random_state is the seed used by the random number generator;
If RandomState instance, random_state is the random number generator;
If None, the random number generator is the RandomState instance used
by `np.random`. Used by svd_solver == 'arpack' or 'randomized'.
```